### PR TITLE
Add additional feature extensions

### PR DESCRIPTION
**Pull request by ShaunaGordon**
I thought including some suggestions for how to customize the Foam workflow more would be useful. I don't think they need to be added to the recommended list in the VSCode settings (though a couple, like Paste Image, might be good candidates), but could be useful for those who want more/other features.

I debated on whether it should be part of this file or in its own, but I'm open to either.

### CODE REVIEW

**Review by riccardoferretti — APPROVED** [review]
sounds good, can I also ask you to include https://marketplace.visualstudio.com/items?itemName=vsls-contrib.gitdoc as you are at it? feel free to merge!

File: docs/recommended-extensions.md [hunk 1]
```diff
@@ -10,3 +10,15 @@ This list is subject to change. Especially the Git ones.
...
+
+These extensions are not (yet?) defined in `.vscode/extensions.json`, but have been used by others and shown to play nice with Foam.
+
+- [Emojisense](https://marketplace.visualstudio.com/items?itemName=bierner.emojisense)
+- [Markdown Emoji](https://marketplace.visualstudio.com/items?itemName=bierner.markdown-emoji) (adds `:smile:` syntax, works with emojisense to provide autocomplete for this syntax)
+- [Mermaid Support for Preview](https://marketplace.visualstudio.com/items?itemName=bierner.markdown-mermaid)
+- [Mermaid Markdown Syntax Highlighting](https://marketplace.visualstudio.com/items?itemName=bpruitt-goddard.mermaid-markdown-syntax-highlighting)
+- [Paste Image](https://marketplace.visualstudio.com/items?itemName=mushan.vscode-paste-image)
+- [VSCode PDF Viewing](https://marketplace.visualstudio.com/items?itemName=tomoki1207.pdf)
+- [Markdown Extended](https://marketplace.visualstudio.com/items?itemName=jebbs.markdown-extended) (with `kbd` option disabled, `kbd` turns wiki-links into non-clickable buttons)
```
Comment (riccardoferretti) [inline docs/recommended-extensions.md:24]:
```suggestion
- [Markdown Extended](https://marketplace.visualstudio.com/items?itemName=jebbs.markdown-extended) (with `kbd` option disabled, `kbd` turns wiki-links into non-clickable buttons)
- [GitDoc](https://marketplace.visualstudio.com/items?itemName=vsls-contrib.gitdoc) (frequent and easy version management via git auto commits)
```

**Review by SanketDG — COMMENTED** [review]
These look great, but I think a better place for these is Recipes, just a flat list of extensions would get really long really fast.

Some of these are already there. Maybe list [[Recipes]] from here?

**Comment by riccardoferretti** [comment]
@SanketDG I agree with the general sentiment of your comment. We have a fairly big task around rethinking documentation (#355), so I am happy in the meantime to take note of these useful extension, to then include them into the new structure. 
Feel free to share your thoughts on that issue if you want to contribute into creating the framework for the next documentation!

**Comment by riccardoferretti** [comment]
@all-contributors add @ShaunaGordon for docs

**Pull request merged**